siphon/simplewebservice/ndbc.py

```diff
@@ -0,0 +1,560 @@
+                     'pressure': 'hPa',
+                     'air_temperature': 'degC',
+                     'water_temperature': 'degC',
+                     'dewpoint': 'degC',
+                     'visibility': 'nautical_mile',
+                     '3hr_pressure_tendency': 'hPa',
+                     'water_level_above_mean': 'feet',
+                     'time': None}
+        resp = endpoint.get_path('data/latest_obs/latest_obs.txt')
+
+        df = pd.read_fwf(StringIO(resp.text), skiprows=2,
+                         na_values='MM', names=col_names)
+        df['time'] = pd.to_datetime(df[['year', 'month', 'day', 'hour', 'minute']], utc=True)
+        df = df.drop(columns=['year', 'month', 'day', 'hour', 'minute'])
+        df.units = col_units
+        return df
+
+    @staticmethod
+    def _check_if_url_valid(url):
+        """Check if a url is valid (returns 200) or not.
+
+        Parameters
+        ----------
+        url : str
+            URL to check
+
+        Returns
+        -------
+            bool if url is valid
+
+        """
+        r = requests.head(url)
+        if r.status_code == 200:
+            return True
+        else:
+            return False
+
+    @classmethod
+    def buoy_data_types(cls, buoy):
+        """Determine which types of data are available for a given buoy.
+
+        Parameters
+        ----------
+        buoy : str
+            Buoy name
+
+        Returns
+        -------
+            dict of valid file extensions and their descriptions
+
+        """
+        endpoint = cls()
+        file_types = {'txt': 'standard meteorological data',
+                      'drift': 'meteorological data from drifting buoys and limited moored'
+                               'buoy data mainly from international partners',
+                      'cwind': 'continuous wind data (10 minute average)',
+                      'spec': 'spectral wave summaries',
+                      'data_spec': 'raw spectral wave data',
+                      'swdir': 'spectral wave data (alpha1)',
+                      'swdir2': 'spectral wave data (alpha2)',
+                      'swr1': 'spectral wave data (r1)',
+                      'swr2': 'spectral wave data (r2)',
+                      'adcp': 'acoustic doppler current profiler',
+                      'ocean': 'oceanographic data',
+                      'tide': 'tide data',
+                      'srad': 'solar radiation data',
+                      'dart': 'water column height',
+                      'supl': 'supplemental measurements data',
+                      'rain': 'hourly rain data'}
+        available_data = {}
+        buoy_url = 'https://www.ndbc.noaa.gov/data/realtime2/' + buoy + '.'
+        for key in file_types:
+            if endpoint._check_if_url_valid(buoy_url + key):
+                available_data[key] = file_types[key]
+        return available_data
+
+    @classmethod
+    def raw_buoy_data(cls, buoy, type='txt'):  # noqa: A002
```

Comment:
How about going with `data_type` instead of `type` so we can avoid overriding a built-in (and the flake8-plugin checking for it).